packages/async-flow/src/template.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @import {HostCall} from './types.js';
+ */
+
+const { assign } = Object;
+
+/** Doesn't need to be exhaustive, just a little prettier than JSON-quoting. */
+const BEST_GUESS_ID_REGEX = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+
+/**
+ * Return an object that mimics a template strings array.
+ *
+ * @param {string[]} strings
+ * @returns {TemplateStringsArray}
+ */
+export const makeTemplateStringsArray = strings =>
+  harden(assign([...strings], { raw: strings }));
```

Comment:
Clever, but the attributes matter. The `raw` should be non-enumerable. How about

```suggestion
export const makeTemplateStringsArray = strings => {
  const result = [...strings];
  defineProperty(result, 'raw', {
    value: strings, // Note: same rather than raw version
    enumerable: false, // not needed, but good to be explicit
  });
  harden(result);
};
```

Reply:
I finessed your suggestion a little bit to avoid needing to coerce `result` with a typecast.